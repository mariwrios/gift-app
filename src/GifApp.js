import React, { useState } from "react"
import AddCategories from "./components/AddCategories"
import GifGrid from "./components/GiftGrid"

const GifApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories)

  /*const handleAdd = () => {
    setCategories([...categories, "Inuyasha"])
  }*/

  return (
    <>
      <h2>GifApp</h2>
      <AddCategories setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map(category => (
          <GifGrid key={category} category={category}></GifGrid>
        ))}
      </ol>
    </>
  )
}

export default GifApp
