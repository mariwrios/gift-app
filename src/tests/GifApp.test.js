import React from "react"
import { shallow } from "enzyme"
import GifApp from "../GifApp"

describe("Probando <GifApp/>", () => {
  test("Que se muestre correctamente <GifApp/>", () => {
    const wrapper = shallow(<GifApp />)
    expect(wrapper).toMatchSnapshot()
  })
  test("Debe de mostrar tantas <GifGridItems/> como categorias tenga", () => {
    const categories = ["inuyasha", "naruto", "SNK"]
    const wrapper = shallow(<GifApp defaultCategories={categories} />)
    expect(wrapper.find("GifGrid").length).toBe(categories.length)
  })
})
