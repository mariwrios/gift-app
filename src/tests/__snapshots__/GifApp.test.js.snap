// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <GifApp/> Que se muestre correctamente <GifApp/> 1`] = `
<Fragment>
  <h2>
    GifApp
  </h2>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
